mrmustard/lab_dev/circuit_components.py

```diff
@@ -497,20 +515,35 @@ def __rshift__(self, other: CircuitComponent) -> CircuitComponent:
         only_bra = not self.wires.ket and not other.wires.ket
         both_sides = self.wires.bra and self.wires.ket and other.wires.bra and other.wires.ket
         if only_ket or only_bra or both_sides:
-            return self @ other
+            ret = self @ other
+            return ret.representation.scalar if len(ret.wires) == 0 else ret
 
         self_needs_bra = (not self.wires.bra) and other.wires.bra and other.wires.ket
         self_needs_ket = (not self.wires.ket) and other.wires.bra and other.wires.ket
         if self_needs_bra or self_needs_ket:
-            return self.adjoint @ (self @ other)
+            ret = self.adjoint @ (self @ other)
+            return ret.representation.scalar if len(ret.wires) == 0 else ret
 
         other_needs_bra = (self.wires.bra and self.wires.ket) and not other.wires.bra
         other_needs_ket = (self.wires.bra and self.wires.ket) and not other.wires.ket
         if other_needs_bra or other_needs_ket:
-            return (self @ other) @ other.adjoint
+            ret = (self @ other) @ other.adjoint
+            return ret.representation.scalar if len(ret.wires) == 0 else ret
 
         raise ValueError(msg)
 
+    def __rrshift__(self, other: Scalar) -> CircuitComponent | np.array:
+        r"""
+        Multiplies a scalar with a circuit component when written as ``scalar >> component``.
+        This is needed when the "component" on the left is the result of a contraction that leaves
```

Comment:
on the left or the right?

Reply:
left.

Reply:
That's why I'm using quotes, it's not a `CircuitComponent`, it's something that doesn't have `__rshift__`. The thing on the right is a `CircuitComponent`.